### DISCUSSION TITLE
SOC HTTPS NAT Issue

### DISCUSSION THREAD

**Discussion by ColeVan** [Security-Onion-Solutions/securityonion#5861] [Unsupported Versions]
Hello everyone,

Maybe someone has experienced this issue before. When wanting to access SOC over https through a NAT'd IP I am unable to do so. Is NGINx purposely configured to only be accessed internally or can NAT rules be setup to access SOC remotely outside I a firewall??

Cole

**Comment by Sahale** [comment]
If it's "clear" NAT (without reverse proxies or something like that) and you sure that it's working, you have to set up routing and firewall on SO system. If it's configured correctly everything should work fine.

**Comment by dougburks** [comment]
Have you tried this?
https://docs.securityonion.net/en/2.3/url-base.html

**Reply by ColeVan** [reply]
There is some sort of NAT issue that I have never encountered before. NAT rules are perfect and work for other services that use HTTPS (443). 
For some reason when Sec Onion Manager tries to NAT back out it will not work. Seems to be some proxy NGINX thing.

**Reply by dougburks** [reply]
To make sure I understand, you installed the box and set its local IP address and then you changed `url_base` to the NAT IP address as shown in https://docs.securityonion.net/en/2.3/url-base.html?

**Reply by ColeVan** [reply]
Thanks for getting back to me.

Currently we have SO2 grid behind a firewall using an internal private IP that is unable to NAT back out of the firewall. All other NAT rules work perfectly fine. Every other service I hit that internally uses https properly NAT's back out. 

For example,

If I want to hit an internal service accessible via https through the firewall I would hit the WAN IP first. Lets say the WAN IP assigned to the firewall is 192.168.1.50. If there are multiple services using https internally on the private IP space of 172.16.0.0/16 behind the firewall and our Sec Onion manager IP for the SOC is https://172.16.0.1 I would need to create specific port on the firewall to allow in for translation so that I properly NAT to the correct internal IP and translate down to the Sec Onion manager.

Example:
From the outside I hit https://192.168.1.50:7373 (allowed throught the firewall) NAT to 172.16.0.1 on port 443. 

On the return Sec Onion will not NAT from 172.16.0.1 port 443 to https:192.168.1.50:7373

Hopefully this gives a bit more context to my current situation. I have not tried the method you suggested. I'm unsure if that will properly fixed the issue. Any suggestions?

Cole

**Reply by dougburks** [reply]
I think that changing `url_base` to the NAT IP address will fix the IP address part of the problem but I suspect it assumes port 443.

**Reply by ColeVan** [reply]
Perhaps your right. I will try and report back if it works. Thanks!

**Reply by objectsecurity** [reply]
I have the same problem. Is there a way to change the port part of the Access URL? If you add the port to the access URL (https://docs.securityonion.net/en/2.3/url-base.html) with colon like this: "100.2.3.4:8888", the updating throws errors. So it will use an external IP or domain correctly but the GUI is only on 443. How can I get this to a different port without having to install another proxy machine? Thank you very much!

**Reply by bit1290** [reply]
Is it possible to have more than one url_base?

**Reply by dougburks** [reply]
@barbosaa99 I've answered your question at https://github.com/Security-Onion-Solutions/securityonion/discussions/7312. In the future, please avoid asking the same question in multiple discussions.
